src/main/kotlin/no/nav/familie/ks/sak/integrasjon/økonomi/internkonsistensavstemming/InternKonsistensavstemmingUtil.kt

```diff
@@ -0,0 +1,110 @@
+    fagsakId: Long
+): Boolean {
+    val utbetalingsperioder =
+        utbetalingsoppdrag?.utbetalingsperiode
+            ?.filter { it.opphør == null }
+            ?: emptyList()
+
+    val forskjellMellomAndeleneOgUtbetalingsoppdraget =
+        hentForskjellIAndelerOgUtbetalingsoppdrag(utbetalingsperioder, andeler)
+
+    when (forskjellMellomAndeleneOgUtbetalingsoppdraget) {
+        is UTBETALINGSPERIODER_UTEN_TILSVARENDE_ANDEL -> secureLogger.info(
+            "Fagsak $fagsakId har sendt utbetalingsperiode(r) til økonomi som ikke har tilsvarende andel tilkjent ytelse." +
+                "\nDet er differanse i perioden(e) ${forskjellMellomAndeleneOgUtbetalingsoppdraget.utbetalingsperioder.tilTidStrenger()}." +
+                "\n\nSiste utbetalingsoppdrag som er sendt til familie-øknonomi på fagsaken er:" +
+                "\n$utbetalingsoppdrag"
+        )
+
+        is INGEN_FORSKJELL -> Unit
+    }
+
+    return forskjellMellomAndeleneOgUtbetalingsoppdraget !is INGEN_FORSKJELL
+}
+
+private fun hentForskjellIAndelerOgUtbetalingsoppdrag(
+    utbetalingsperioder: List<Utbetalingsperiode>,
+    andeler: List<AndelTilkjentYtelse>
+): AndelOgOppdragForskjell {
+    val utbetalingsperioderUtenTilsvarendeAndel = utbetalingsperioder.filter {
+        it.erIngenPersonerMedTilsvarendeAndelITidsrommet(andeler)
+    }
+
+    return if (utbetalingsperioderUtenTilsvarendeAndel.isEmpty()) {
+        INGEN_FORSKJELL
+    } else {
+        UTBETALINGSPERIODER_UTEN_TILSVARENDE_ANDEL(utbetalingsperioderUtenTilsvarendeAndel)
+    }
+}
+
+private fun Utbetalingsperiode.erIngenPersonerMedTilsvarendeAndelITidsrommet(
+    andeler: List<AndelTilkjentYtelse>
+): Boolean {
+    val andelsTidslinjerPerPersonOgYtelsetype = andeler
+        .groupBy { Pair(it.aktør, it.type) }
+        .map { (_, andeler) -> andeler.tilBeløpstidslinje() }
+
+    return andelsTidslinjerPerPersonOgYtelsetype.all {
+        !this.harTilsvarendeAndelerForPersonOgYtelsetype(it)
+    }
+}
+
+private fun Utbetalingsperiode.harTilsvarendeAndelerForPersonOgYtelsetype(
+    andelerTidslinjeForEnPersonOgYtelsetype: Tidslinje<BigDecimal>
+): Boolean {
+    val erAndelLikUtbetalingTidslinje = this.tilBeløpstidslinje()
+        .kombinerMed(andelerTidslinjeForEnPersonOgYtelsetype) { utbetalingsperiode, andel ->
+            utbetalingsperiode?.let { utbetalingsperiode == andel }
+        }
+
+    return erAndelLikUtbetalingTidslinje.tilPerioder().all { it.verdi != false }
+}
+
+private fun Utbetalingsperiode.tilBeløpstidslinje() =
+    listOf(
+        Periode(
+            fom = this.vedtakdatoFom.førsteDagIInneværendeMåned(),
+            tom = this.vedtakdatoTom.sisteDagIMåned(),
+            verdi = this.sats
+        )
+    ).tilTidslinje()
+
+private fun List<AndelTilkjentYtelse>.tilBeløpstidslinje() =
+    this.map {
+        Periode(
+            fom = it.stønadFom.toLocalDate().førsteDagIInneværendeMåned(),
+            tom = it.stønadTom.toLocalDate().sisteDagIMåned(),
```

Comment:
Er det lettere å bare bruke [atDay](https://developer.android.com/reference/kotlin/java/time/YearMonth#atday) og [atEndOfMonth](https://developer.android.com/reference/kotlin/java/time/YearMonth#atendofmonth
)?
```suggestion
            fom = it.stønadFom.atDay(1),
            tom = it.stønadTom.atEndOfMonth(),
```